File: API-Patio-El-Olvidado/src/PatioElOlvidado.Domain/Entities/Producto.cs
namespace PatioElOlvidado.Domain.Entities;

public class Producto
{
    public int Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string? Descripcion { get; set; }
    public decimal Precio { get; set; }
    public string Categoria { get; set; } = string.Empty;
    /// <summary>URL de imagen (sin upload binario en MVP).</summary>
    public string? Imagen { get; set; }
    /// <summary>Etiquetas separadas por coma (ej. "vegano,sin gluten").</summary>
    public string? Etiquetas { get; set; }
    public bool Activo { get; set; } = true;
}

File: Patio-El-Olvidado/src/types/producto.ts
export interface Producto {
  id: number
  nombre: string
  descripcion: string | null
  precio: number
  categoria: string
  imagen: string | null
  etiquetas: string | null
  activo: boolean
}

export interface ProductoFilter {
  q?: string
  categoria?: string
  etiqueta?: string
  soloActivos?: boolean
}

export interface ProductoPayload {
  nombre: string
  descripcion?: string | null
  precio: number
  categoria: string
  imagen?: string | null
  etiquetas?: string | null
  activo: boolean
}
